cai_causal_graph/time_series_causal_graph.py

```diff
@@ -1279,23 +1289,45 @@ def to_numpy_by_lag(self) -> Tuple[Dict[int, numpy.ndarray], List[str]]:
         assert graph.variables is not None
         return adjacency_matrices, graph.variables
 
+    def add_node_to_cache(self, node: TimeSeriesNode):
+        """Add a node to the model caches."""
+        self.lag_to_nodes[node.time_lag].append(node)
+        self.variable_name_to_nodes[node.variable_name].append(node)
+
+    def remove_node_from_cache(self, node: TimeSeriesNode):
+        """Remove a node to the model caches."""
+        # Handle removal from lag-to-nodes cache
+        if node.time_lag in self.lag_to_nodes and node in self.lag_to_nodes[node.time_lag]:
```

Comment:
I'm thinking we can assume that the node we are removing exists in the cache. Therefore we can remove the if statements and just do the `remove` calls? Would be much more efficient as we don't have to search through the lists multiple times. Maybe we just wrap the remove statements in a try/except that raises some meaningful error if for whatever reason it isn't found